plugins/hls-code-range-plugin/src/Ide/Plugin/CodeRange.hs

```diff
@@ -126,6 +148,61 @@ findPosition pos root = go Nothing root
             startOfRight <- _start . _codeRange_range <$> V.headM right
             if pos < startOfRight then binarySearchPos left else binarySearchPos right
 
+-- | Traverses through the code range and it children to a folding ranges.
+--
+-- It starts with the root node, converts that into a folding range then moves towards the children.
+-- It converts each child of each root node and parses it to folding range and moves to its children.
+findFoldingRanges :: CodeRange -> [FoldingRange]
+findFoldingRanges r@(CodeRange _ children _) =
+    let frChildren :: [FoldingRange] = concat $ V.toList $ fmap findFoldingRanges children
+    in case createFoldingRange r of
+        Just x  -> x:frChildren
+        Nothing -> []
```

Comment:
This should e `frChildren`! Did the tests keep passing with this? If they did, you are missing some tests!

Reply:
I’ll test it manually as well. There were no errors in the codebase as well as the tests.

Reply:
Well, that tells us the case you're not testing: a code range with a kind that we can't handle, but which has children that could provide valid folding ranges. That's the case that will go wrong her.

Reply:
So CodeRangeKind can only be -->
```
-- | 'CodeKind' represents the kind of a code range
data CodeRangeKind =
    -- | ordinary code
    CodeKindRegion
    -- | the group of imports
  | CodeKindImports
  -- | a comment
  | CodeKindComment
    deriving (Show, Eq, Generic, NFData)
```

I am not sure how can I mention something that does not have a valid codekind.

Reply:
Then why is there a `Nothing` case in `crkToFrk`? (https://github.com/haskell/haskell-language-server/pull/3058/files#diff-b72658b4ae69940ecdda95504f965e4ac60e70b0c7582360149b9bb52bd361f6R201) I guess that's actually a redundant match and it could be total. `crkToFrk` returning `Nothing` is the case that would trip this up.

Reply:
Yes makes sense, I’ll remove this case